EIPS/eip-3224.md

```diff
@@ -0,0 +1,466 @@
+- Many other things, deplnding on your environment
+
+### Distinguished Signed Data
+
+Applications implementing this EIP to sign message data should
+ensure there are no collisions within the data which could
+result in ambiguously signed data.
+
+@TODO: Expand on this; compare packed data to ABI encoded data?
+
+### Enumeration
+
+If an abort occurs during signing, the response from this call
+should match the response from a declined signing request;
+otherwise this could be used for enumeration attacks, etc. A
+random interactive-scale delay should also be added, otherwise
+a < 10ms response could be interpreted as an error.
+
+### Replayablility
+
+Transactions contain an explicit nonce, but signed messages do
+not.
+
+For many purposes, such as signing in, a nonce could be
+injected (using block.timestamp) into the data. The log in
+service can verify this is a recent timestamp. The timestamp
+may or may not be omitted from the description string in this
+case, as it it largely useful internally only.
+
+In general, when signing messages a nonce often makes sense to
+include to prevent the same signed data from being used in the
+future.
+
+
+## Copyright
+
+Copyright and related rights waived via [CC0](https://creativecommons.org/publicdomain/zero/1.0/).
+
+----
+
+# Draft notes and thoughts
+
+Below are some notes to move above, discuss or otherwise. This will be removed
+or moved above to meaningful sections
+
+**NOTES:**
+
+- Clients should perform addditional due diligence when describing addresses, for example via reverse records
+- Hardware wallets can be fed the required results from eth_getProof along with the describer and display on-device the expected operation
+- I am assuming there will be an invalid byte as an opcode that starts a eWASM so that those fail on EVM, and are identifable on compatible engines; this should allow this to be backwards and forwards compatible too
+- While Solidity is not awesome at string manipulation, there is nothing precluding a new compiler which simplifies string operations that compiles to bytecode or more advanced string libraries.
+- For popular legacy contracts, bytecode can be baked into wallets, or deployed on chain with the describer address baked in
+- When signing described data, the `to` address must be entangled in the signature to prevent replaying. In the case of transactions, this is implicitly handled by the `to` in the transaction. In messages, this is handled by including the to address in the EIP-191 *version specific data*. This is similar to the domain in [EIP-712](./eip-712.md), but allows for the dapp developer to define their own replay strategy (if any is desired)
+- Localization is quite important; there should be a way to encode this... This can be a separate EIP
+- Should a simple Markdown be supported? Keep in mind chain data must then be markdown-escaped.
+
+**Desired Features:**
+
+- Simple
+- re-invent as few wheels as possible
```

Comment:
```suggestion
```

EIPs should be written toward their final form, and this section isn't a valid EIP section so should be removed.  If you want, you can mark this PR as a review so people don't start reading it, and then you can delete this before submitting to be merged as a draft.  Alternatively, consider moving this content to something like HackMD or the discussions-to link.